src/MasterSite.php

```diff
@@ -1153,4 +1154,69 @@ public function dismiss_dashboard_notice(): void {
 
 		wp_die( 'Notice dismissed.', 200 );
 	}
+
+	/**
+	 * Adds functionality to Twig.
+	 *
+	 * @param \Twig\Environment $twig The Twig environment.
+	 * @return \Twig\Environment
+	 */
+	public function p4_optimize_img_url( $twig ) {
+		$options             = get_option( 'planet4_options' );
+		$cf_img_optimization = $options['cloudflare_img_opt'] ?? false;
+		$cf_options_txt      = $options['cloudflare_options_txt'] ?? '';
+
+		// Add Twig filter(cf_img_url) which convert image url/s into cloudflare optimized image url/s.
+		$twig->addFilter(
+			new Twig_SimpleFilter(
+				'cf_img_url',
+				function ( $source, $srcset = '' ) use ( $cf_img_optimization, $cf_options_txt ) {
+					if ( ! $cf_img_optimization ) {
```

Comment:
could we move this outside of this call, maybe before the `$twig->addFilter` ? It seems like if no optimization is selected in global options, we don't have to add this filter at all

Reply:
if we skip the `$twig->addFilter` part, the frontend where we use `cf_img_url` would be invalid and through's error.